x-pack/plugins/fleet/server/services/epm/packages/remove.ts

```diff
@@ -243,7 +244,7 @@ async function deleteIndexTemplate(esClient: ElasticsearchClient, name: string):
 
 async function deleteComponentTemplate(esClient: ElasticsearchClient, name: string): Promise<void> {
   // '*' shouldn't ever appear here, but it still would delete all templates
-  if (name && name !== '*') {
+  if (name && name !== '*' && name !== FLEET_ECS_MAPPINGS_COMPONENT_TEMPLATE_NAME) {
```

Comment:
@nchaulet and @felixbarny. what do you think?

Reply:
Looking at https://github.com/elastic/kibana/pull/163731, there's an array for `STACK_COMPONENT_TEMPLATES`. Could you confirm whether or not these stack component templates are susceptible of being uninstalled by fleet? If not, it's probably because of this:

https://github.com/elastic/kibana/blob/6fea1ace880e03cb38d4dad0195b136ce2659b3a/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts#L621-L622

Which means that it would be enough to add the `ecs@mapping` component template into the `STACK_COMPONENT_TEMPLATES` array, without further filtering in `deleteComponentTemplate`.

Reply:
I have tested what you are suggesting with the version of the stack `8.13.0-yufkxnwm-SNAPSHOT` since we need 8.13 to be compatible with Kibana from main and a version old enough that it doesn't contain the ES fix. 

The ES bug is there but it doesn't try to remove the component template `ecs@mappings` or any other component template that is not installed by the integration (which are `*@package` and `*@custom`).

I have committed some changes.

can@felixbarny and @nchaulet and @kpollich have a look please?